pkg/controller/kubelet-config/kubelet_config_bootstrap.go

```diff
@@ -0,0 +1,165 @@
+		if kubeletConfig == nil {
+			continue
+		}
+		if err := validateUserKubeletConfig(kubeletConfig); err != nil {
+			return nil, newForgetError(err)
+		}
+	}
+
+	userDefinedSystemReserved := make(map[string]string)
+	var res []*mcfgv1.MachineConfig
+	for _, kubeletConfig := range kubeletConfigs {
+		if kubeletConfig == nil || kubeletConfig.Spec == (mcfgv1.KubeletConfigSpec{}) {
+			continue
+		}
+		for _, pool := range mcpPools {
+			// skip the pool if no default label for kubeletconfig
+			_, hasMasterLabel := pool.Labels[kubeletCfgMasterLable]
+			_, hasWorkerLabel := pool.Labels[kubeletCfgWorkerLable]
+			if !hasMasterLabel && !hasWorkerLabel {
+				continue
+			}
+			role := pool.Name
+
+			originalKubeConfig, err := generateOriginalKubeletConfigWithFeatureGates(controllerConfig, templateDir, role, features)
+			if err != nil {
+				return nil, err
+			}
+			if kubeletConfig.Spec.TLSSecurityProfile != nil {
+				// Inject TLS Options from Spec
+				observedMinTLSVersion, observedCipherSuites := getSecurityProfileCiphers(kubeletConfig.Spec.TLSSecurityProfile)
+				originalKubeConfig.TLSMinVersion = observedMinTLSVersion
+				originalKubeConfig.TLSCipherSuites = observedCipherSuites
+			}
+
+			kubeletIgnition, logLevelIgnition, autoSizingReservedIgnition, err := generateKubeletIgnFiles(kubeletConfig, originalKubeConfig, userDefinedSystemReserved)
+			if err != nil {
+				return nil, err
+			}
+
+			tempIgnConfig := ctrlcommon.NewIgnConfig()
+			if autoSizingReservedIgnition != nil {
+				tempIgnConfig.Storage.Files = append(tempIgnConfig.Storage.Files, *autoSizingReservedIgnition)
+			}
+			if logLevelIgnition != nil {
+				tempIgnConfig.Storage.Files = append(tempIgnConfig.Storage.Files, *logLevelIgnition)
+			}
+			if kubeletIgnition != nil {
+				tempIgnConfig.Storage.Files = append(tempIgnConfig.Storage.Files, *kubeletIgnition)
+			}
+
+			rawIgn, err := json.Marshal(tempIgnConfig)
+			if err != nil {
+				return nil, err
+			}
+			managedKey, err := generateSuffixManagedKeyKubelet(pool, kubeletConfig, generatedManagedKeys)
+			if err != nil {
+				return nil, err
+			}
+			var (
+				mc *mcfgv1.MachineConfig
+				ok bool
+			)
+			mc, ok = generatedMCs[managedKey]
+			if !ok {
+				mc, err = ctrlcommon.MachineConfigFromIgnConfig(role, managedKey, kubeletIgnition)
+				if err != nil {
+					return nil, err
+				}
+				generatedMCs[managedKey] = mc
+			}
+
+			mc.Spec.Config.Raw = rawIgn
+			mc.SetAnnotations(map[string]string{
+				ctrlcommon.GeneratedByControllerVersionAnnotationKey: version.Hash,
+			})
+			oref := metav1.OwnerReference{
+				APIVersion: controllerKind.GroupVersion().String(),
+				Kind:       controllerKind.Kind,
+			}
+			mc.SetOwnerReferences([]metav1.OwnerReference{oref})
+			res = append(res, mc)
+		}
+	}
+	return res, nil
+}
+
+func generateSuffixManagedKeyKubelet(pool *mcfgv1.MachineConfigPool, cfg *mcfgv1.KubeletConfig, managedKeys map[string]string) (string, error) {
```

Comment:
could you add a comment here about this func

Reply:
Fixed